### PR TITLE
006 Ordering of `dplyr::group_by()`

### PR DESCRIPTION
**Pull request by DavisVaughan**

### CODE REVIEW

File: 006-dplyr-group-by-ordering.Rmd [hunk 1]
```diff
@@ -0,0 +1,161 @@
...
+
+    -   <https://github.com/r-lib/vctrs/pull/1441>
+
+-   dplyr PR for converting to `vec_locate_sorted_groups()`
+
+    -   TODO
+
+## Backwards Compatibility
+
+It is difficult to know how much "in the wild" user code will break as a result of changing from using the system locale to the C locale in `group_by()`. Ideally, users won't be relying on row order for most of their analysis scripts, but we are empathetic to the fact that this is not always the case. Our plan is to run reverse dependency checks and help fix any package code with issues that arise from this, and to provide a pre-release blog post noting this change so that users can update their code as required.
```
Comment (hadley) [inline 006-dplyr-group-by-ordering.Rmd:136]: Do we need to provide a helper to make it easier for affected users to switch back to the old behaviour?

**Review by hadley — COMMENTED** [review]
Looks good to kick off further discussion to me. 

I do wonder if we should consider a bigger re-thinking of grouping, by providing a `.by` argument to every dplyr verb. That would allow bigger changes to grouping without affecting existing code.

**Comment by DavisVaughan** [comment]
> providing a .by argument to every dplyr verb

I wonder how users would provide the `.drop` argument to `group_by()`

File: 006-dplyr-group-by-ordering.Rmd (hunk 1 above)
Comment (DavisVaughan) [reply 006-dplyr-group-by-ordering.Rmd:136]: We could potentially provide a global option like `dplyr.legacy_group_by_locale = TRUE` that would use the old behavior.

I imagine providing a function like `group_by_legacy()` would be just as annoying as adding an `arrange()` call to actually fix it, so I don't think we'd go that route.

Plus the global option way let's you control group-by calls that you might not control directly.

Ideally when we'd introduce this we'd mention that this is a temporary measure that we'd faze out fairly quickly?

Comment (DavisVaughan) [reply 006-dplyr-group-by-ordering.Rmd]: I've added the short-term global option, `dplyr.legacy_group_by_locale`, to the proposal

**Review by hadley — APPROVED** [review]

**Pull request merged**